src/main/java/com/github/redouane59/twitter/TwitterClient.java

```diff
@@ -399,6 +404,43 @@ public boolean hideReply(final String tweetId, final boolean hide) {
         return result;
     }
 
+    @Override
+    public void startFilteredStream(Consumer<ITweet> consumer){
+        String      url    = this.urlHelper.getFilteredStreamUrl();
+        this.requestHelperV2.executeGetRequestWithConsumer(url, consumer);
+    }
+
+    @Override
+    public List<StreamRule> retrieveFilteredStreamRules() {
+        String       url    = this.urlHelper.getFilteredStreamRulesUrl();
+        StreamRulesDTO result = this.requestHelperV2.executeGetRequest(url, StreamRulesDTO.class).orElseThrow(NoSuchElementException::new);
+        return result.getData();
+    }
+
+    @Override
+    public StreamRule addFilteredStreamRule(String value, String tag){
+        String       url    = this.urlHelper.getFilteredStreamRulesUrl();
+        try {
+            StreamRule rule = StreamRule.builder().value(value).tag(tag).build();
+            String body = "{\"add\": ["+TwitterClient.OBJECT_MAPPER.writeValueAsString(rule)+"]}";
+            StreamRulesDTO result = this.requestHelperV2.executePostRequest(url, body, StreamRulesDTO.class).orElseThrow(NoSuchElementException::new);
+            if(result.getData()==null || result.getData().size()==0) throw new IllegalArgumentException();
+            return result.getData().get(0);
+        }
+        catch (JsonProcessingException e) {
+            LOGGER.error(e.getMessage(), e);
+            throw new IllegalArgumentException();
```

Comment:
Hmmmm i'm not really confortable with it.
I tried to create this : `Try<String> bodyTry = Try.of(TwitterClient.OBJECT_MAPPER.writeValueAsString(rule));`  but it is still telling me that the exception is not managed in the code...
2nd remark : I hate the fact to have to explicitly write the try catch (or whatever) whereas I'm sure that it will never fail, I just construct the object the line before to get its string representation. I don't think that any case could fail.